packages/snap/src/UI.service.ts

```diff
@@ -0,0 +1,171 @@
+import { QueryVCsRequestResult } from '@blockchain-lab-um/masca-types';
+import {
+  Component,
+  copyable,
+  divider,
+  heading,
+  panel,
+  spinner,
+  text,
+} from '@metamask/snaps-ui';
+import { W3CVerifiableCredential } from '@veramo/core';
+
+import StorageService from './storage/Storage.service';
+
+class UIService {
+  static origin: string;
+
+  static async init(origin: string) {
+    this.origin = origin;
+  }
+
+  static async snapConfirm(
+    content: Component,
+    force?: boolean
+  ): Promise<boolean> {
+    const state = StorageService.get();
+
+    const { disablePopups, friendlyDapps } = state.snapConfig.dApp;
+
+    // Show popups if force is true or if popups are not disabled AND the dApp is not friendly
+    if (force || !(disablePopups || friendlyDapps.includes(this.origin))) {
+      const res = await snap.request({
+        method: 'snap_dialog',
+        params: {
+          type: 'confirmation',
+          content,
+        },
+      });
+      return res as boolean;
+    }
+    return true;
+  }
+
+  static async snapAlert(content: Component, force?: boolean): Promise<void> {
+    const state = StorageService.get();
+
+    const { disablePopups, friendlyDapps } = state.snapConfig.dApp;
+
+    // Show popups if force is true or if popups are not disabled AND the dApp is not friendly
+    if (force || !(disablePopups || friendlyDapps.includes(this.origin))) {
+      await snap.request({
+        method: 'snap_dialog',
+        params: {
+          type: 'alert',
+          content,
+        },
+      });
+    }
+  }
+}
+
+export const querySpinnerContent = () =>
+  panel([heading('Gathering Credentials...'), spinner()]);
+
+export const queryAllContent = (vcs: QueryVCsRequestResult[]) =>
+  panel([
+    heading('Share Verifiable Credentials'),
+    text(
+      'Would you like to share some/all Verifiable Credentials with this dApp?'
+    ),
+    divider(),
+    text(`**Total number of VCs is ${vcs.length.toString()}**`),
+    divider(),
+    text(
+      `You can disable this popup from appearing again in Masca Settings on Masca.io.`
+    ),
+  ]);
+
+export const saveCredentialContent = (
+  store: string | string[],
+  verifiableCredential: any
+) =>
+  panel([
+    heading('Save Verifiable Credential'),
+    text('Would you like to save the following Verifiable Credential?'),
+    divider(),
+    text(`Credential:`),
+    copyable(JSON.stringify(verifiableCredential, null, 2)),
+    text(
```

Comment:
Maybe move this before the listed credential so the user doesn't need to scroll to see where the credential will be save.